### ISSUE TITLE
rernaker eval 

### ISSUE THREAD

**Issue by riyajatar37003** [embeddings-benchmark/mteb#1230]
Hi,
how can evaluate the reranker style model using mteb?
this is my custom model 
`
import torch
from typing import cast, List, Union, Tuple
from transformers import  AutoTokenizer, AutoModelForSequenceClassification
import sys

class Reranker:
    def __init__(
            self,
            model_name_or_path: str = None,
            use_fp16: bool = False
    ) -> None:

        self.tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
        self.model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path,trust_remote_code=True)#,num_hidden_layers=16,num_attention_heads=16)
        print(self.model)
        if torch.cuda.is_available():
            self.device = torch.device('cuda:0')
        elif torch.backends.mps.is_available():
            self.device = torch.device('mps')
        else:
            self.device = torch.device('cpu')
            use_fp16 = False
        if use_fp16:
            self.model.half()

        self.model = self.model.to(self.device)

        self.model.eval()

        self.num_gpus = torch.cuda.device_count()
        if self.num_gpus > 1:
            print(f"----------using {self.num_gpus}*GPUs----------")
            self.model = torch.nn.DataParallel(self.model)

    @torch.no_grad()
    def compute_score(self, sentence_pairs: Union[List[Tuple[str, str]], Tuple[str, str]], batch_size: int = 256,
                      max_length: int = 512) -> List[float]:
        if self.num_gpus > 0:
            batch_size = batch_size * self.num_gpus

        assert isinstance(sentence_pairs, list)
        if isinstance(sentence_pairs[0], str):
            sentence_pairs = [sentence_pairs]

        all_scores = []
        for start_index in range(0, len(sentence_pairs), batch_size):
            sentences_batch = sentence_pairs[start_index:start_index + batch_size]
            # sentences_batch = [["query: "+p[0],"passage: "+p[1]] for p in sentences_batch]
            inputs = self.tokenizer(
                sentences_batch,
                padding=True,
                truncation=True,
                return_tensors='pt',
                max_length=max_length,
            ).to(self.device)

            scores = self.model(**inputs, return_dict=True).logits[:,0]#.logits.view(-1, ).float() #.logits[:,0]#
            all_scores.extend(scores.cpu().numpy().tolist())

        if len(all_scores) == 1:
            return all_scores[0]
        return all_scores

`

**Comment by Muennighoff** [comment]
There's a snippet for eval'ing rerankers here: https://github.com/embeddings-benchmark/mteb?tab=readme-ov-file#using-a-cross-encoder-for-reranking

It seems you have a custom model definition, which you may be able to get working using something similar to here https://github.com/embeddings-benchmark/mteb?tab=readme-ov-file#using-a-custom-model - Maybe if you figure it out, you could help update the docs with a `Using a custom reranker model` section!

**Comment by isaac-chung** [comment]
Closing this now. Feel free to reopen if there's anything unclear.
